### DISCUSSION TITLE
Custom Form & File Uploads

### DISCUSSION THREAD

**Discussion by duncanmcclean** [filamentphp/filament#10342] [Help]
I've created a [custom form component](https://filamentphp.com/docs/3.x/forms/getting-started#overview) and I'm trying to add a `FileUpload` file to the form's schema. 

The form displays correctly but when I upload any files, I get [the following error](https://flareapp.io/share/Lm8qX1MP) upon upload:

```
Property type not supported in Livewire for property: [null]
```

I'm relatively new to Filament & Livewire stuff so I could very possibly be doing something wrong. Here's a very simple version of my component so you can reproduce:

```php
<?php

namespace App\Filament\Pages;

use Filament\Forms;
use Filament\Forms\Form;
use Filament\Pages\Page;

class SendEmail extends Page implements Forms\Contracts\HasForms
{
    use Forms\Concerns\InteractsWithForms;

    protected static string $view = 'send-email';

    public $attachment;

    public function form(Form $form): Form
    {
        return $form->schema([
            Forms\Components\FileUpload::make('attachment')
                ->disk('uploads'),
        ]);
    }
}
```

**Comment by ShaungBhone** [comment]
What does the `->disk('uploads')` mean? Is that s3?

**Reply by duncanmcclean** [reply]
It's just pointing to the `uploads` filesystem I have configured. This error happens without this line too.

**Reply by ShaungBhone** [reply]
Need code details. Where is `statePath()` and model and `mount()`?

**Reply by duncanmcclean** [reply]
I don't have either of those methods in my component. They aren't in the docs for [creating forms](https://filamentphp.com/docs/3.x/forms/getting-started#overview) or [Filament pages](https://filamentphp.com/docs/3.x/panels/pages#creating-a-page).

**Reply by ShaungBhone** [reply]
What are you supposed to do? I've never seen a before page together with form!

**Reply by ShaungBhone** [reply]
Please check here https://github.com/jeffgreco13/filament-breezy/blob/2.x/src/Pages/TwoFactorPage.php

**Reply by duncanmcclean** [reply]
I'm building a page where my client can send emails to their customers, with a file attachment. 

From the looks of things in the documentation, you can create a Filament Form on any Livewire component (of which custom pages are).

**Comment by duncanmcclean** [comment]
I seem to have managed to get it working... looks like `public $attachment` needed to be typehinted as an array. Now that it's an array, I'm no longer getting any errors.

Not entirely sure *why* that's the case but it's working.

**Reply by ShaungBhone** [reply]
Glad to help.

**Reply by duncanmcclean** [reply]
Thanks for your help! Hope you have a great rest of your weekend!

**Reply by pr4xx** [reply]
For anyone stumbling over this: setting array as an typehint is not the correct solution. The author forgot to add:
```
public function mount(): void
{
    $this->form->fill([]);
}
```
This is mandatory according to the filament docs.
